base/linalg/dense.jl

```diff
@@ -362,6 +362,39 @@ function inv{T}(A::StridedMatrix{T})
     return convert(typeof(parent(Ai)), Ai)
 end
 
+"""
+    factorize(A)
+
+Compute a convenient factorization of `A`, based upon the type of the input matrix.
+`factorize` checks `A` to see if it is symmetric/triangular/etc. if `A` is passed
+as a generic matrix. `factorize` checks every element of `A` to verify/disqualify
+each property. The return value can be reused for efficient solving of multiple
+systems. For example: `A=factorize(A); x=A\\b; y=A\\C`.
+
+| Properties of `A`          | type of factorization                          |
+|:---------------------------|:-----------------------------------------------|
+| Positive-definite          | Cholesky (see [`cholfact`](:func:`cholfact`))  |
+| Dense Symmetric/Hermitian  | Bunch-Kaufman (see [`bkfact`](:func:`bkfact`)) |
+| Sparse Symmetric/Hermitian | LDLt (see [`ldltfact`](:func:`ldltfact`))      |
+| Triangular                 | Triangular                                     |
+| Diagonal                   | Diagonal                                       |
+| Bidiagonal                 | Bidiagonal                                     |
+| Triidiagonal               | LU (see [`lufact`](:func:`lufact`))            |
+| (Hermitian) tridiagonal    | LDLt (see [`ldltfact`](:func:`ldltfact`))      |
```

Comment:
Does Hermitian tridiagonal here refer to both complex Hermitian matrices and real symmetric matrices, or the former exclusive of the latter? If inclusive, then isn't Hermitian tridiagonal the standard language? And complex symmetric tridiagonal the standard language for the symmetry under strict (nonconjugate) transpose of a complex matrix, understood to not fall under the Hermitian classification?